tests/integration.rs

```diff
@@ -51,17 +56,14 @@ fn integration_test() {
         .expect("unable to run clippy");
 
     let stderr = String::from_utf8_lossy(&output.stderr);
-    if stderr.contains("internal compiler error") {
-        let backtrace_start = stderr
-            .find("thread 'rustc' panicked at")
-            .expect("start of backtrace not found");
-        let backtrace_end = stderr
-            .rfind("error: internal compiler error")
+    if let Some(backtrace_start) = stderr.find("error: internal compiler error") {
+        let backtrace_end = stderr[backtrace_start..]
+            .find("end of query stack")
             .expect("end of backtrace not found");
 
         panic!(
             "internal compiler error\nBacktrace:\n\n{}",
-            &stderr[backtrace_start..backtrace_end]
+            &stderr[backtrace_start..backtrace_start + backtrace_end + 18]
```

Comment:
I think this could have done with a `static` and using its `.len()` here instead of a magic number, but this is a very minor niggle.

Reply:
I really should have. I actually got the number wrong the first time I wrote it.